src/ts/InternalBitmovinYospacePlayer.ts

```diff
@@ -237,11 +238,17 @@ export class InternalBitmovinYospacePlayer implements BitmovinYospacePlayerAPI {
           this.session = session;
 
           this.calculateAdParts();
-          // clone source to not modify passed object
-          const clonedSource = {
-            ...source,
-            hls: this.session.getPlaybackUrl(), // use received url from yospace
-          };
+          const clonedSource = source.hls
+            ? {
+                ...source,
+                hls: this.session.getPlaybackUrl(), // use received url from yospace
+                dash: undefined as string,
+              }
+            : {
+                ...source,
+                dash: this.session.getPlaybackUrl(), // use received url from yospace
+                hls: undefined as string,
```

Comment:
I don't think the typecasting to string should be needed here, just undefined should work looking at the types:

```suggestion
                dash: undefined,
              }
            : {
                ...source,
                dash: this.session.getPlaybackUrl(), // use received url from yospace
                hls: undefined,
```